addons/mrp/tests/test_byproduct.py

```diff
@@ -479,3 +479,18 @@ def test_03_check_byproducts_update(self):
         mo.button_mark_done()
         self.assertEqual(len(mo.move_byproduct_ids.move_line_ids), 2)
         self.assertEqual(mo.move_byproduct_ids.product_id, mo.move_byproduct_ids.move_line_ids.product_id)
+
+    def test_create_mrp_production_with_context(self):
+        """
+        Test creating MRP production while passing default value
+        for 'move_type' in context.
+        """
+        mp = self.env['mrp.production'].with_context(default_move_type="out_invoice").create({
```

Comment:
I don't really like this, it does not make sense to provide such a context 
It's always best to write use cases that can be reproduced from the webclient
By calling `action_create_invoice`, you will create the invoice and have the problematic context

Reply:
(It is probably possible to adapt the test so you don't need the mrp part and can then write the test in `sale_subscription_stock`)